src/main/java/org/apache/maven/reporting/MavenReport.java

```diff
@@ -53,7 +53,9 @@ public interface MavenReport {
     void generate(Sink sink, Locale locale) throws MavenReportException;
 
     /**
-     * Get the base name used to create report's output file(s).
+     * Get the output name denoting a base location relative to the {@link #getReportOutputDirectory()}
+     * used to create the report's main output file. The base location may contain path components
```

Comment:
delete "The base location may contain path components
     * to better structure the report output". as it's not relevant to this method. Possibly the interaction of the methods could be described in the class level comment

Reply:
@elharo, again I ask you to read that from a user's perspective. Users need more context information than providers and profit from examples and explanations that a provider might correctly deem redundant. A certain degree of redundancy or context helps humans to understand things.

Reply:
From a user's perspective "to better structure the report output" adds nothing. Would anyone expect it to be "to worse structure the report output?" If not, then this means nothing.It's not redundant. It doesn't add or repeat anything.

Reply:
Well, I did not write the text, so I am not clinging to it. Maybe I am just a bit less judgemental or opinionated about it. You are probably right insofar as the javadoc can do without that part and still be understood.

But just like in requirements engineering it helps the implementor to not just understand what she needs to deliver and what the acceptance criteria are, the "(in order) to ..." subclause in a user story adds context and explains the all-important business value. One could argue: If the story has no value, why would the creator prioritise it into the sprint in the first place? So it adds nothing. But wait a minute, does it not? Yes, it does. As an agile coach, since 2006 I have seen literally hundreds of value-less user stories, so it does helpto explain not just the existence of or requirement for something, but also its purpose or value.

Reply:
Specific actionable context is useful, but too much "business need" is just wasted words. Commons examples:

* in order to improve performance
* in order to improve customer service
* in order to better serve you

None of these actually explain anything. Here don't just say "to better structure the report output" but explain how the report output structure is changed, or say nothing at all. I don't know if this is accurate but something like "to divide the report documents into separate subdirectories" would be more concrete. "to better structure the report output" is not a user story and does not explain the business value.

Reply:
@elharo, I agree with you in this case.